doc/user/methods/mean_shift.md

```diff
@@ -0,0 +1,341 @@
+mlpack::MeanShift<false> ms(25.0, 100 /* max iterations */);
+
+arma::mat centroids;
+arma::Row<size_t> assignments;
+ms.Cluster(dataset, assignments, centroids);
+
+// Print the number of clusters and the number of points in each cluster.
+std::cout << "MeanShift found " << centroids.n_cols << " clusters."
+    << std::endl;
+for (size_t i = 0; i < centroids.n_cols; ++i)
+{
+  std::cout << " - Cluster " << i << " has " << arma::accu(assignments == i)
+      << " points assigned to it." << std::endl;
+}
+```
+
+---
+
+Perform mean shift clustering with the triangular kernel on the cloud dataset,
+using 32-bit floating point matrices to represent the data.
+
+```c++
+// See https://datasets.mlpack.org/cloud.csv.
+arma::fmat dataset;
+mlpack::data::Load("cloud.csv", dataset, true);
+
+// Create the MeanShift object using a TriangularKernel.
+mlpack::TriangularKernel tk;
+mlpack::MeanShift<true, mlpack::TriangularKernel> ms(50.0 /* radius */,
+                                                     1000 /* max iterations */,
+                                                     tk);
+
+// Perform clustering.
+arma::fmat centroids;
+arma::Row<size_t> assignments;
+ms.Cluster(dataset, assignments, centroids);
+
+// Print the number of clusters and the number of points in each cluster.
+std::cout << "MeanShift found " << centroids.n_cols << " clusters."
+    << std::endl;
+for (size_t i = 0; i < centroids.n_cols; ++i)
+{
+  std::cout << " - Cluster " << i << " has " << arma::accu(assignments == i)
+      << " points assigned to it." << std::endl;
+}
+```
+
+---
+
+### Advanced Functionality: Template Parameters
+
+The `MeanShift` class has two template parameters that can be used for custom
+behavior.  The full signature of the class is:
+
+```
+MeanShift<UseKernel, KernelType>
+```
+
+ * `UseKernel` (default `true`) is a `bool` parameter representing whether a
+   kernel function is used to weight points during centroid recomputation.  If
+   it is `false`, then each point within distance `radius` of the centroid will
+   be used (without weighting) to recompute the centroid.  This strategy (with
+   `UseKernel = false`) is also known as using a 'flat kernel'.
+
+ * `KernelType` represents the kernel function (or Parzen window) to be used to
+   weight points during centroid recomputation.  Although many
+   [mlpack kernels](../core.md#kernels) are available, only those with
+   `Gradient()` functions (described below) are supported.  Available kernels
+   for drop-in usage include:
+   - [`GaussianKernel`](../core.md#gaussiankernel) *(default)*
+   - [`EpanechnikovKernel`](../core.md#epanechnikovkernel)
+   - [`LaplacianKernel`](../core.md#laplaciankernel)
+   - [`SphericalKernel`](../core.md#sphericalkernel) *(note: this is equivalent
+     to the flat kernel, or, setting `UseKernel = false`)*
+   - [`TriangularKernel`](../core.md#triangularkernel)
+
+Custom kernels can be easily implemented, and must implement only one function
```

Comment:
```suggestion
Custom kernels for mean shift can be easily implemented, and must implement only one function
```
because in other cases we might require `Evaluate()` but it might sound repetitive just a suggestion from my side.

Reply:
Agreed, I think it's good to be specific here 👍